src/cascadia/TerminalApp/Jumplist.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "Jumplist.h"
+
+#include <Propkey.h>
+
+using namespace winrt::TerminalApp;
+
+//  This property key isn't already defined in propkey.h, but is used by UWP Jumplist to determine the icon of the jumplist item.
+//  IShellLink's SetIconLocation isn't going to read "ms-appx://" icon paths, so we'll need to use this to set the icon.
+DEFINE_PROPERTYKEY(PKEY_AppUserModel_DestListLogoUri, 0x9F4C2855, 0x9F79, 0x4B39, 0xA8, 0xD0, 0xE1, 0xD4, 0x2D, 0xE1, 0xD5, 0xF3, 29);
+#define INIT_PKEY_AppUserModel_DestListLogoUri                                             \
+    {                                                                                      \
+        { 0x9F4C2855, 0x9F79, 0x4B39, 0xA8, 0xD0, 0xE1, 0xD4, 0x2D, 0xE1, 0xD5, 0xF3 }, 29 \
+    }
+
+// Method Description:
+// - Updates the items of the Jumplist based on the given settings.
+// Arguments:
+// - settings - The settings object to update the jumplist with.
+// Return Value:
+// - <none>
+void Jumplist::UpdateJumplist(std::shared_ptr<::TerminalApp::CascadiaSettings> settings)
+{
+    winrt::com_ptr<ICustomDestinationList> jumplistInstance;
+    if (FAILED(CoCreateInstance(CLSID_DestinationList, nullptr, CLSCTX_ALL, IID_PPV_ARGS(&jumplistInstance))))
+    {
+        return;
+    }

```

Comment:
Can you use `RETURN_IF_FAILED` here or is there a macro we have for these?

Reply:
Honestly I could make the function return `HR` so I could use that macro, and just have the caller ignore the return if they want.